.github/workflows/publish.yml

```diff
@@ -31,6 +31,9 @@ jobs:
       with:
         node-version: 16
 
+    - name: Configure npm
+      run: npm config set //wombat-dressing-room.appspot.com/:_authToken=$NODE_AUTH_TOKEN
```

Comment:
I think you also need to add this:

```
- env:
      NODE_AUTH_TOKEN: ${{ secrets.RELEASE_BACKED_NPM_TOKEN }}
```

to this step in order to convert the github secret into an environment variable for this command to use. based on https://docs.npmjs.com/using-private-packages-in-a-ci-cd-workflow#set-the-token-as-an-environment-variable-on-the-cicd-server

Reply:
I think the [env is scoped to the specific step](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsenv). So I think it needs to be in the publish step where it currently is. See also this example: https://github.com/azu/lerna-monorepo-github-actions-release/blob/master/.github/workflows/publish.yml

Reply:
But you aren't currently setting the environment variable `NODE_AUTH_TOKEN` anywhere. so you need to set that somewhere. and since you need that variable in this step I think you would need to set it in this step. but you can also set it for the whole workflow or job if we need that.

in other words from your example I don't think you are doing this line https://github.com/azu/lerna-monorepo-github-actions-release/blob/315ed6abbf54698b62fc6885b1d425d0a6259053/.github/workflows/publish.yml#L32

what you are doing with the `environment` in the `publish` step is making that secret available to use in this workflow but you're not actually using it anywhere that i can see.

Reply:
My understanding was that the environment variable doesn't actually get evaluated until the `npm publish` step. I don't think it would be evaluated during `npm config` since you don't *have* to call `npm config` (if you're using a static `.npmrc` file).

[This example](https://github.com/azu/lerna-monorepo-github-actions-release/blob/315ed6abbf54698b62fc6885b1d425d0a6259053/.github/workflows/publish.yml#L82) that uses a static `.npmrc` sets the environment variable in the publish step. [This example ](https://stackoverflow.com/a/57652197) that double-quote `echo`s to the `.npmrc` sets it in the config step and the publish step. So idk what to believe.

I'm cool moving it and trying it out.